src/sentry/plugins/providers/integration_repository.py

```diff
@@ -27,14 +27,43 @@ def __init__(self, id):
 
     def dispatch(self, request, organization, **kwargs):
         try:
+            # get data via API (so we know it's up to date)
             config = self.get_repository_data(organization, request.DATA)
+
+        except Exception as e:
+            return self.handle_api_error(e)
+
+        else:
+            # check to see if we already have this repository, and update any
+            # data if applicable
+            repo = Repository.objects.filter(
+                organization_id=organization.id,
+                provider=self.id,
+                external_id=config.get('external_id')
+            ).first()
+            if repo is not None:
```

Comment:
Just style here. In python people tend to write:
```py
if repo:
```
since anything that is `None` will evaluate to false in python.

Reply:
Yup, I know... and that specific thing caused a bug in one of our SDK's just last week. :-)

I've been writing it `if repo:` for years, too. But in light of that bug, I'm finally taking my python instructor's advice from years ago and trying to start being explicit and not relying on falsiness (sp?). 

But ¯\_(ツ)_/¯ - I'm happy to put it back if it's the Sentry Way.

Reply:
You just have the be cognizant of when it can give you a false positive. So, if you think `0` or` ''` or `[]` or `{}` could be things you would want to accept, then it's ok to just go ahead and say what you mean. But in the case here, repo can be either a Repository model object or in the case you are filtering, then it could be an empty QuerySet. First is giving you a couple of things for free here, one being returning `None` if nothing is returned. So in the case you had originally the only two return options are the model object or None. So, it's very safe to simply say `if repo`.